src/plugins/types.ts

```diff
@@ -0,0 +1,6 @@
+import { AnyFunc, WrapWithName } from '../types'
+
+export interface PluginConfig {
+  matchers: WrapWithName<AnyFunc>[]
```

Comment:
Using AnyFunc here is also unhelpful. We could enforce that the function returns a matcher instance